### PR TITLE
Requirements module stubs

### PR DESCRIPTION
**Pull request by ianrowan**
## Link to Issue
Closes: #4958 

## Description of Changes
- Creates detailed `Requirements` type
- Creates `ValidateGroupMembership.ts` module
- Creates `validateRequirements`

## Test Plan
Stubs only

## Other Considerations

- Can be merged for use on route development 
- Requirements type can be used for db model column(move the file if necessary) on whichever ticket will implement that

### CODE REVIEW

File: packages/commonwealth/server/util/requirementsModule/requirmentsTypes.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  chain_id: string;
+  token_symbol: string;
+};
+type ThresholdData = {
+  threshold: string;
+  source: ContractSource | NativeSource | CosmosSource;
+};
+
+type AllowlistData = {
+  allow: LengthConstrainedString<42>[];
```
Comment (jnaviask) [inline packages/commonwealth/server/util/requirementsModule/requirmentsTypes.ts:27]: Not sure we can make this guarantee re address length, as cosmos addresses are in bech32 format and could be longer than 42 chars.

File: packages/commonwealth/server/util/requirementsModule/requirmentsTypes.ts
Comment (jnaviask) [inline packages/commonwealth/server/util/requirementsModule/requirmentsTypes.ts:1]: misspelled filename?

**Review by jnaviask — CHANGES_REQUESTED** [review]

File: packages/commonwealth/server/util/requirementsModule/requirmentsTypes.ts (hunk 1 above)
Comment (ianrowan) [reply packages/commonwealth/server/util/requirementsModule/requirmentsTypes.ts]: Changed to string for now

File: packages/commonwealth/server/util/requirementsModule/requirmentsTypes.ts
Comment (ianrowan) [reply packages/commonwealth/server/util/requirementsModule/requirmentsTypes.ts:1]: fixed

**Review by jnaviask — APPROVED** [review]
Approving, but tagging @rbennettcw as reviewer to ensure interfaces are sufficient for his needs in building the API.

**Review by rbennettcw — APPROVED** [review]
Interface looks good

**Pull request merged**